src/charm.py

```diff
@@ -161,13 +375,8 @@ def _on_update_status(self, _) -> None:
             if state == "restarting":
                 # Restart the stuck replica.
                 self._restart_postgresql_service()
-            elif state == "stopping":
-                # Change needed to force an immediate failover.
-                self._patroni.change_master_start_timeout(0)
-                # Restart the stuck previous leader (will trigger an immediate failover).
-                self._restart_postgresql_service()
-                # Revert configuration to default.
-                self._patroni.change_master_start_timeout(None)
+            elif state == "starting" or state == "stopping":
+                self.force_primary_change()
```

Comment:
This is needed because a bug on Pebble. It'll be removed along with other functions in the next PR as the current version of Juju solved the problem of zombie process from PostgreSQL not being cleaned by Pebble.

Reply:
This is needed because a bug on Pebble. It'll be removed along with other functions in the next PR as the current version of Juju solved the problem of zombie process from PostgreSQL not being cleaned by Pebble.